src/pages/ProfilePage.js

```diff
@@ -0,0 +1,237 @@
+import compose from '../libs/compose';
+import CommunicationsLink from '../components/CommunicationsLink';
+import Tooltip from '../components/Tooltip';
+import CONST from '../CONST';
+import * as ReportUtils from '../libs/ReportUtils';
+import * as Expensicons from '../components/Icon/Expensicons';
+import MenuItem from '../components/MenuItem';
+import AttachmentModal from '../components/AttachmentModal';
+import PressableWithoutFocus from '../components/PressableWithoutFocus';
+import * as Report from '../libs/actions/Report';
+import OfflineWithFeedback from '../components/OfflineWithFeedback';
+import AutoUpdateTime from '../components/AutoUpdateTime';
+import * as UserUtils from '../libs/UserUtils';
+import * as PersonalDetails from '../libs/actions/PersonalDetails';
+import FullScreenLoadingIndicator from '../components/FullscreenLoadingIndicator';
+import BlockingView from '../components/BlockingViews/BlockingView';
+import * as Illustrations from '../components/Icon/Illustrations';
+import variables from '../styles/variables';
+import ROUTES from '../ROUTES';
+
+const matchType = PropTypes.shape({
+    params: PropTypes.shape({
+        /** accountID passed via route /a/:accountID */
+        accountID: PropTypes.string,
+
+        /** report ID passed */
+        reportID: PropTypes.string,
+    }),
+});
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The personal details of all users */
+    personalDetails: personalDetailsPropType,
+
+    /** Route params */
+    route: matchType.isRequired,
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Date login was validated, used to show info indicator status */
+        validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    // When opening someone else's profile (via deep link) before login, this is empty
+    personalDetails: {},
+    loginList: {},
+};
+
+/**
+ * Gets the phone number to display for SMS logins
+ *
+ * @param {Object} details
+ * @param {String} details.login
+ * @param {String} details.displayName
+ * @returns {String}
+ */
+const getPhoneNumber = (details) => {
+    // If the user hasn't set a displayName, it is set to their phone number, so use that
+    const displayName = lodashGet(details, 'displayName', '');
+    const parsedPhoneNumber = parsePhoneNumber(displayName);
+    if (parsedPhoneNumber.possible) {
+        return parsedPhoneNumber.number.e164;
+    }
+
+    // If the user has set a displayName, get the phone number from the SMS login
+    return details.login ? Str.removeSMSDomain(details.login) : '';
+};
+
+function ProfilePage(props) {
+    const accountID = lodashGet(props.route.params, 'accountID', 0);
+
+    useEffect(() => {
+        PersonalDetails.openPublicProfilePage(accountID);
```

Comment:
It could be worth it to not call this command if accountID is <= 0 or something right?